solver/solver.go

```diff
@@ -14,9 +15,26 @@ import (
 
 // Stats holds the stats related to a Solve.
 type Stats struct {
-	CreateOps int
-	UpdateOps int
-	DeleteOps int
+	CreateOps *AtomicInt32Counter
+	UpdateOps *AtomicInt32Counter
+	DeleteOps *AtomicInt32Counter
+}
+
+type AtomicInt32Counter struct {
+	counter int32
+	lock    sync.RWMutex
+}
+
+func (a *AtomicInt32Counter) Increment(delta int32) {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+	a.counter += delta
+}
+
+func (a *AtomicInt32Counter) Read() int32 {
```

Comment:
nit - `Count` might be a better name because it describes the returned value better, and because `Read` is already in use by `io.Reader`.